controllers/util/solr_util.go

```diff
@@ -214,16 +214,20 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 			},
 		}
 	} else {
-		emptyDirVolume := corev1.Volume{
-			Name: solrDataVolumeName,
-			VolumeSource: corev1.VolumeSource{
-				EmptyDir: &corev1.EmptyDirVolumeSource{},
-			},
+		ephemeralVolume := corev1.Volume{
+			Name:         solrDataVolumeName,
+			VolumeSource: corev1.VolumeSource{},
 		}
 		if solrCloud.Spec.StorageOptions.EphemeralStorage != nil {
```

Comment:
If `solrCloud.Spec.StorageOptions.EphemeralStorage` is empty, but not `nil`, then I think we will have an empty VolumeSource. We want to default to a plain EmptyDir in this case as well.

Reply:
I might have misunderstood what you meant, but I've added a check for if EmptyDir is nil, and a corresponding test. There was already a test for if EphemeralStorage is empty though (if that's what you meant?):
https://github.com/apache/solr-operator/blob/c78b2b80ab097cb7f1b573f4fca8903ba7f63eb9/controllers/solrcloud_controller_storage_test.go#L240

Reply:
Ahh wonderful. Must have missed the test. But good change on the code 👍